hs-bindgen/src-internal/HsBindgen/Frontend/Pass/Parse/Monad.hs

```diff
@@ -160,6 +186,31 @@ checkHasMacroExpansion extent = do
       , any (\e -> fromMaybe False (rangeContainsLoc range e)) expansions
       ]
 
+recordSource :: CXCursor -> M ()
+recordSource curr = do
+    mNamespace <- dispatch curr $ \case
+      CXCursor_InclusionDirective -> return Nothing
```

Comment:
Let's add a catch-all, with a comment: 

We intentionally do selection as part of parsing, rather than a separate step: if the user does not select certain declarations (perhaps because they live deep in the bowels of some system libraries), we also do not need to parse them. Since `recordSource` is called on _all_ declarations, selected or not, we must ensure that we don't error out on such unsupported cases.